api/src/main/java/com/epam/pipeline/manager/cluster/performancemonitoring/ResourceMonitoringManager.java

```diff
@@ -221,16 +267,18 @@ private void processIdleRuns(final List<PipelineRun> runs) {
         final IdleRunAction action = IdleRunAction.valueOf(preferenceManager
                 .getPreference(SystemPreferences.SYSTEM_IDLE_ACTION));
 
+        final List<PipelineRun> runsToUpdateTags = new ArrayList<>(running.size());
```

Comment:
move updating tags and sending notifications to `processRuns` method, it would be more readable and also logical